Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/dialogController/NavigationController.java

```diff
@@ -207,4 +241,68 @@ protected void share(FolderType selectedFolder) {
 			Messagebox.show(e.getMessage(), APROMORE, Messagebox.OK, Messagebox.ERROR);
 		}
 	}
+
+
+	public void rename(FolderType selectedFolder) throws InterruptedException {
+		
+			if (selectedFolder == null) {
+				Notification.error(Labels.getLabel("portal_failedFind_message"));
+				return;
+			}
+		
+		try {
+			boolean canChange = false;
+			try {
+				User currentUser = mainC.getSecurityService().getUserById(UserSessionManager.getCurrentUser().getId());
+				canChange = ItemHelpers.canModify(currentUser, selectedFolder);
+			} catch (Exception e) {
+				Notification.error(e.getMessage());
+				return;
+			}
+
+			if (canChange) {
+				mainC.eraseMessage();
+				new RenameFolderController(mainC, selectedFolder.getId(), selectedFolder.getFolderName());
+			} else {
+				Notification.error(Labels.getLabel("portal_noPrivilegeRename_message"));
+			}
```

Comment:
The control flow seems clunky here.  You should be able to rearrange it so that there's an `if (ItemHelpers.canModify(currentUser, selectedFolder) { ... }` block rather than needing the `canChange` variable.